File: cpp/g20.base/Student.cpp
#include "Student.h"

namespace PersonSpace {

  /******* CONSTRUCTORS *******/
  Student::Student(const char* n)           : Person(n) { /* nothing */ }
  Student::Student(const char* n, double g) : Person(n) { setGPA(g); }

  /******* MUTATORS (SETTERS) *******/
  void   Student::setGPA(double g) { gpa = g; }
  /******* ACESSORS (GETTERS) *******/
  double Student::getGPA()         { return gpa; }

}

File: ruby/g01.inheritance/Student.rb
#!/usr/bin/ruby
# class declaration
class Student < Person
  ###### ACCESSORS (SETTERS) ######
  def setGPA (gpa)
    @gpa = gpa         # set instance data
  end

  ###### ACCESSORS (GETTERS) ######
  def getGPA
    @gpa               # return instance data
  end
end
